django/db/models/sql/query.py

```diff
@@ -1137,7 +1137,29 @@ def build_filter(self, filter_expr, branch_negated=False, current_negated=False,
 
         # Work out the lookup type and remove it from the end of 'parts',
         # if necessary.
-        value, lookups, used_joins = self.prepare_lookup_value(value, lookups, can_reuse, allow_joins)
+        if isinstance(value, (list, tuple)) and value:
+            klass = type(value)
+            new_value = []
+            new_lookups = []
+            new_used_joins = []
+            # Look up using the iterable of arguments first, in case it's an empty list
+            for inner_value in value:
+                inner_value, lookups, used_joins = self.prepare_lookup_value(inner_value,
+                    lookups,
+                    can_reuse,
+                    allow_joins)
+                new_value.append(inner_value)
+                for lookup in lookups:
+                    if lookup not in new_lookups:
+                        new_lookups.append(lookup)
```

Comment:
I believe this to be broken. It shouldn't be possible for one of the rhs values to produce a new lookup which others didn't produce.